Backend/Core/Services/OrderService.cs

```diff
@@ -310,5 +310,38 @@ public async Task ConfirmDeliveryAsync(string userId, int orderId)
                 await _orderRepository.UpdateAsync(order);
             }
         }
+
+        public async Task RejectDeliveryAsync(string userId, int orderId)
+        {
+            var user = await _userRepository.GetByIdAsync(userId);
+
+            ExtensionMethods.UserNullCheck(user);
+
+            var userRole = await _identityRoleService.GetUserRoleAsync(user);
+
+            if (userRole == IdentityRoleNames.User.ToString())
+            {
+                var order = await _orderRepository.SingleOrDefaultAsync(
+                    new OrderSpecification.GetByClientConfirmedDelivery(userId, orderId));
+
+                ExtensionMethods.OrderNullCheck(order);
+
+                order.IsAcceptedByClient = false;
+
+                await _orderRepository.UpdateAsync(order);
+
+            }
+            else if (userRole == IdentityRoleNames.Courier.ToString())
+            {
+                var order = await _orderRepository.SingleOrDefaultAsync(
+                    new OrderSpecification.GetByCourierConfirmedDelivery(userId, orderId));
+
+                ExtensionMethods.OrderNullCheck(order);
+
+                order.IsAcceptedByCourier = false;
+
+                await _orderRepository.UpdateAsync(order);
+            }
+        }
```

Comment:
This is a pure copy-paste (except for changing "true" to "false" and 2 specifications) from the previous method, merge them and do not repeat. 
And give an error that the customer or courier cannot cancel the confirmation of the order because it is not confirmed yet